{
  try {
    latch.await(1000,MILLISECONDS);
  }
 catch (  InterruptedException e) {
    throw new RuntimeException("await failed",e);
  }
  if (throwable != null) {
    if (throwable instanceof Error) {
      throw (Error)throwable;
    }
 else     if (throwable instanceof RuntimeException) {
      throw (RuntimeException)throwable;
    }
  }
}
